sdccc/src/test/java/com/draeger/medical/sdccc/tests/glue/invariant/InvariantSubscriptionHandlingTestTest.java

```diff
@@ -0,0 +1,708 @@
+
+        testClass = new InvariantSubscriptionHandlingTest();
+        testClass.setup();
+    }
+
+    @AfterEach
+    void testDown() throws TimeoutException {
+        baseMarshalling.stopAsync().awaitTerminated(DEFAULT_TIMEOUT);
+        marshalling.stopAsync().awaitTerminated(DEFAULT_TIMEOUT);
+        storage.close();
+    }
+
+    /**
+     * Test whether calling the test without any input data causes the test to fail.
+     */
+    @Test
+    public void testRequirementR0056NoTestData() {
+        assertThrows(NoTestData.class, testClass::testRequirementR0056);
+    }
+
+    /**
+     * Checks if the test fails with a NoTestData exception if only description modification reports with
+     * modification type delete were seen during the test run.
+     *
+     * @throws Exception on any exception
+     */
+    @Test
+    public void testRequirementR0056NoTestData2() throws Exception {
+        final var metric = mdibBuilder.buildNumericMetric(
+                NUMERIC_HANDLE,
+                MetricCategory.CLC,
+                MetricAvailability.CONT,
+                mdibBuilder.buildCodedValue("Millibar"),
+                BigDecimal.ONE);
+        final var part = buildDescriptionModificationReportPart(
+                DescriptionModificationType.DEL,
+                PARENT_HANDLE,
+                new ImmutablePair<>(metric.getLeft(), metric.getRight()));
+        final var report = buildDescriptionModificationReport(MdibBuilder.DEFAULT_SEQUENCE_ID, BigInteger.ZERO, part);
+
+        messageStorageUtil.addInboundSecureHttpMessage(storage, report);
+        assertThrows(NoTestData.class, testClass::testRequirementR0056);
+    }
+
+    /**
+     * Checks if the test fails with a NoTestData exception if only description modification reports without
+     * states were seen during the test run.
+     *
+     * @throws Exception on any exception
+     */
+    @Test
+    public void testRequirementR0056NoTestData3() throws Exception {
+        final var metric = mdibBuilder.buildNumericMetric(
+                NUMERIC_HANDLE,
+                MetricCategory.CLC,
+                MetricAvailability.CONT,
+                mdibBuilder.buildCodedValue("Millibar"),
+                BigDecimal.ONE);
+        final var part = buildDescriptionModificationReportPart(
+                DescriptionModificationType.CRT,
+                PARENT_HANDLE,
+                new ImmutablePair<>(metric.getLeft(), metric.getRight()));
+
+        // clear states from report part
+        part.getState().clear();
+
+        final var report = buildDescriptionModificationReport(MdibBuilder.DEFAULT_SEQUENCE_ID, BigInteger.ZERO, part);
+
+        messageStorageUtil.addInboundSecureHttpMessage(storage, report);
+        assertThrows(NoTestData.class, testClass::testRequirementR0056);
+    }
+
+    /**
+     * Tests whether the test passes when the description modification report where sent before any report
```

Comment:
I'm afraid there is still a typo in there.
```suggestion
     * Tests whether the test passes when the description modification report were sent before any report
```

Reply:
fixed